documentation/library_guide/kernel_templates/esimd/radix_sort.rst

```diff
@@ -0,0 +1,322 @@
++------------------------------------------------------+------------------------------------------------------------------+
+|                                                      | The sequences(s) of elements to apply the algorithm to.          |
+|  - ``rng`` (1)                                       | They can be provided as:                                         |
+|  - ``first``, ``last`` (2)                           |                                                                  |
+|  - ``keys_rng``, ``vals_rng`` (3)                    | - ``sycl::buffer`` (1,3),                                        |
+|  - ``keys_first``, ``keys_last``, ``vals_first`` (4) | - ``oneapi::dpl::experimental::ranges::views::all`` (1,3)        |
+|                                                      | - ``oneapi::dpl::experimental::ranges::views::subrange`` (1,3)   |
+|                                                      | - USM pointer (2,4)                                              |
+|                                                      | - ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` (2,4)          |
++------------------------------------------------------+------------------------------------------------------------------+
+|  ``param``                                           | Kernel configuration structure. ``data_per_workitem``,           |
+|                                                      | can be any value among ``32``, ``64``, ``96``,..., ``k * 32``;   |
+|                                                      | ``workgroup_size`` can be only ``64``.                           |
++------------------------------------------------------+------------------------------------------------------------------+
+
+
+Return Value
+------------
+
+``sycl::event`` object representing a status of the algorithm execution.
+
+
+Memory Usage
+------------
+
+.. _local-memory:
+
+Local Memory
+~~~~~~~~~~~~
+
+The local memory is allocated as shown in the pseudo-code blocks below:
+
+- ``radix_sort`` (1,2):
+
+  .. code:: python
+
+     ranks = 2 * (2 ^ radix_bits) * workgroup_size + (2 * workgroup_size) + 4 * (2 ^ radix_bits)
+     reorder = sizeof(key_type) * data_per_workitem * workgroup_size + 4 * (2 ^ radix_bits)
+     allocated_bytes = round_up_to_nearest_multiple(max(ranks, reorder), 2048)
+
+
+- ``radix_sort_by_key`` (3,4):
+
+  .. code:: python
+
+     ranks = 2 * (2 ^ radix_bits) * workgroup_size + (2 * workgroup_size) + 4 * (2 ^ radix_bits)
+     reorder = (sizeof(key_type) + sizeof(value_type)) * data_per_workitem * workgroup_size + 4 * (2 ^ radix_bits)
+     allocated_bytes = round_up_to_nearest_multiple(max(ranks, reorder), 2048)
+
+The device must have enough local memory to execute the selected configuration.
+
+
+Global Memory
+~~~~~~~~~~~~~
+
+The global (USM device) memory is allocated as shown in the pseudo-code blocks below:
```

Comment:
I think a similarly verbose description similar to my comment about local memory above may be helpful.

Also, we use `..._bytes` for both of the intermediate "variables" here, but not above, perhaps we should add `_bytes` above for consistency.

Reply:
Thanks, done